### ISSUE TITLE
"Got request to deoptimize un-deoptimizable method"

### ISSUE THREAD

**Issue by M66B** [rovo89/Xposed#325]
Xposed version 90-beta3 on a Nexus 5X with Android 8.1 Oreo

These seem to conflict with each other:
https://github.com/M66B/XPrivacyLua/blob/master/app/src/main/java/eu/faircode/xlua/XLua.java#L464
https://github.com/M66B/XPrivacyLua/blob/master/app/src/main/java/eu/faircode/xlua/XLua.java#L817

```
02-03 17:47:20.480  8952  8952 I XLua.Xposed: Dynamic invoke public synchronized void com.android.webview.chromium.ContentSettingsAdapter.setUserAgentString(java.lang.String)
02-03 17:47:20.481  8952  8952 W zygote64: Got request to deoptimize un-deoptimizable method org.luaj.vm2.Varargs org.luaj.vm2.LuaValue.invoke(org.luaj.vm2.LuaValue[])
02-03 17:47:20.481  8952  8952 F zygote64: quick_exception_handler.cc:326] Check failed: callee_method_ != nullptr org.luaj.vm2.LuaValue.invoke
02-03 17:47:20.680  8952  8952 F zygote64: runtime.cc:528] Runtime aborting...
02-03 17:47:20.680  8952  8952 F zygote64: runtime.cc:528] Dumping all threads without appropriate locks held: thread list lock
02-03 17:47:20.680  8952  8952 F zygote64: runtime.cc:528] All threads:
02-03 17:47:20.680  8952  8952 F zygote64: runtime.cc:528] DALVIK THREADS (39):
02-03 17:47:20.680  8952  8952 F zygote64: runtime.cc:528] "main" prio=7 tid=1 Runnable
02-03 17:47:20.680  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=0 dsCount=0 flags=2 obj=0x73d04b30 self=0x76d0cbea00
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8952 nice=-4 cgrp=default sched=0/0 handle=0x77557d69a8
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | state=R schedstat=( 1511899118 94552139 902 ) utm=133 stm=17 core=5 HZ=100
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | stack=0x7fe0cb6000-0x7fe0cb8000 stackSize=8MB
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | held mutexes= "abort lock" "mutator lock"(shared held)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 00000000003cf938  /system/lib64/libart.so (art::DumpNativeStack(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, int, BacktraceMap*, char const*, art::ArtMethod*, void*)+208)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 00000000004a3d0c  /system/lib64/libart.so (art::Thread::DumpStack(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, bool, BacktraceMap*, bool) const+348)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 00000000004bbb50  /system/lib64/libart.so (art::DumpCheckpoint::Run(art::Thread*)+884)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 00000000004b43fc  /system/lib64/libart.so (art::ThreadList::RunCheckpoint(art::Closure*, art::Closure*)+480)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 00000000004b3ba0  /system/lib64/libart.so (art::ThreadList::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, bool)+288)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #05 pc 000000000048b3f0  /system/lib64/libart.so (art::AbortState::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char>>&) const+220)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #06 pc 000000000047a9d0  /system/lib64/libart.so (art::Runtime::Abort(char const*)+196)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #07 pc 0000000000572a5c  /system/lib64/libart.so (android::base::LogMessage::~LogMessage()+996)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #08 pc 000000000046f0e0  /system/lib64/libart.so (art::DeoptimizeStackVisitor::FinishStackWalk()+332)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #09 pc 000000000046ec14  /system/lib64/libart.so (art::DeoptimizeStackVisitor::VisitFrame()+84)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #10 pc 0000000000492eb4  /system/lib64/libart.so (_ZN3art12StackVisitor9WalkStackILNS0_16CountTransitionsE0EEEvb+1088)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #11 pc 000000000046e01c  /system/lib64/libart.so (art::QuickExceptionHandler::DeoptimizeStack()+96)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #12 pc 000000000052146c  /system/lib64/libart.so (art::artDeoptimizeImpl(art::Thread*, art::DeoptimizationKind, bool)+608)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #13 pc 0000000000521208  /system/lib64/libart.so (artDeoptimize+16)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #14 pc 000000000055b7c8  /system/lib64/libart.so (art_quick_deoptimize+168)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #15 pc 0000000000032230  /dev/ashmem/dalvik-jit-code-cache (deleted) (???)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at org.luaj.vm2.LuaValue.invoke(LuaValue.java:1789)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at eu.faircode.xlua.XLua$4$1.execute(XLua.java:464)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at eu.faircode.xlua.XLua$4$1.afterHookedMethod(XLua.java:430)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:375)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.webkit.WebView.<init> [XposedHooked](<Xposed>:-2)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.webkit.WebView.<init> [XposedOriginal](WebView.java:571)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native method)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:361)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.webkit.WebView.<init> [XposedHooked](<Xposed>:-2)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.webkit.WebView.<init> [XposedOriginal](WebView.java:558)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native method)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:361)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.webkit.WebView.<init> [XposedHooked](<Xposed>:-2)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.webkit.WebView.<init> [XposedOriginal](WebView.java:548)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native method)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:361)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.webkit.WebView.<init> [XposedHooked](<Xposed>:-2)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.webview.q.<init>(:com.google.android.gms.DynamiteModulesA@11975440:1)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.webview.n.call(:com.google.android.gms.DynamiteModulesA@11975440:6)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.util.cl.a(:com.google.android.gms.DynamiteModulesA@11975440:23)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.webview.k.a(:com.google.android.gms.DynamiteModulesA@11975440:5)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.js.c.<init>(:com.google.android.gms.DynamiteModulesA@11975440:9)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.js.m.run(:com.google.android.gms.DynamiteModulesA@11975440:7)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.util.al.a(:com.google.android.gms.DynamiteModulesA@11975440:377)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.js.l.a(:com.google.android.gms.DynamiteModulesA@11975440:19)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.js.l.a(:com.google.android.gms.DynamiteModulesA@11975440:29)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   - locked <0x04b84db3> (a java.lang.Object)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.js.function.b.b(:com.google.android.gms.DynamiteModulesA@11975440:4)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.ab.a(:com.google.android.gms.DynamiteModulesA@11975440:55)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.ab.a(:com.google.android.gms.DynamiteModulesA@11975440:4)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.av.a(:com.google.android.gms.DynamiteModulesA@11975440:55)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.internal.client.ay.onTransact(:com.google.android.gms.DynamiteModulesA@11975440:43)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.os.Binder.transact(Binder.java:627)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.b.je$a$a.a(unavailable:-1)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.b.jo.a(unavailable:-1)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   - locked <0x0597cf70> (a java.lang.Object)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.i.a(unavailable:-1)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.google.android.gms.ads.i.a(unavailable:-1)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.anu.main.myandroid.MainActivity.onCreate(unavailable:-1)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.app.Activity.performCreate(Activity.java:6999)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.app.Activity.performCreate(Activity.java:6990)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.app.ActivityThread.-wrap11(ActivityThread.java:-1)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.os.Handler.dispatchMessage(Handler.java:106)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.os.Looper.loop(Looper.java:164)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at android.app.ActivityThread.main(ActivityThread.java:6494)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at java.lang.reflect.Method.invoke(Native method)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:108)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528] "Jit thread pool worker thread 0" prio=7 tid=2 Native
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d40000 self=0x76c860e000
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8955 nice=9 cgrp=default sched=0/0 handle=0x76c85ff4f0
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 770440208 63243171 443 ) utm=68 stm=8 core=4 HZ=100
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76c8501000-0x76c8503000 stackSize=1021KB
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait_queue_me+0xe0/0x148
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait+0x104/0x224
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: do_futex+0xe0/0x928
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_futex+0x11c/0x1b0
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000001d4ac  /system/lib64/libc.so (syscall+28)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 00000000000e8a94  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+152)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 00000000004bd2f8  /system/lib64/libart.so (art::ThreadPool::GetTask(art::Thread*)+256)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 00000000004bc9f0  /system/lib64/libart.so (art::ThreadPoolWorker::Run()+124)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 00000000004bc4c4  /system/lib64/libart.so (art::ThreadPoolWorker::Callback(void*)+148)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #05 pc 0000000000067d0c  /system/lib64/libc.so (__pthread_start(void*)+36)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #06 pc 000000000001eba4  /system/lib64/libc.so (__start_thread+68)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   (no managed stack frames)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528] "Signal Catcher" prio=7 tid=3 WaitingInMainSignalCatcherLoop
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d41420 self=0x76d0cbf400
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8957 nice=0 cgrp=default sched=0/0 handle=0x76c84fe4f0
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 244115 265625 5 ) utm=0 stm=0 core=5 HZ=100
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76c8404000-0x76c8406000 stackSize=1005KB
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: do_sigtimedwait+0xf0/0x1e4
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_rt_sigtimedwait+0xe0/0x13c
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000006a5d8  /system/lib64/libc.so (__rt_sigtimedwait+8)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 0000000000028468  /system/lib64/libc.so (sigwait+60)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 00000000004929c4  /system/lib64/libart.so (art::SignalSet::Wait()+44)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 00000000004924d8  /system/lib64/libart.so (art::SignalCatcher::WaitForSignal(art::Thread*, art::SignalSet&)+248)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 0000000000490d04  /system/lib64/libart.so (art::SignalCatcher::Run(void*)+268)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #05 pc 0000000000067d0c  /system/lib64/libc.so (__pthread_start(void*)+36)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #06 pc 000000000001eba4  /system/lib64/libc.so (__start_thread+68)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   (no managed stack frames)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528] "ReferenceQueueDaemon" prio=7 tid=4 Waiting
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d414a8 self=0x76d0cbfe00
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8959 nice=4 cgrp=default sched=0/0 handle=0x76baceb4f0
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 1421148 2034790 16 ) utm=0 stm=0 core=3 HZ=100
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76babe9000-0x76babeb000 stackSize=1037KB
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait_queue_me+0xe0/0x148
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait+0x104/0x224
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: do_futex+0xe0/0x928
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_futex+0x11c/0x1b0
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000001d4ac  /system/lib64/libc.so (syscall+28)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 00000000000e8a94  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+152)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 00000000003cabb8  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+636)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 00000000003cc628  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+376)
02-03 17:47:20.681  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 000000000000070c  /system/framework/arm64/boot.oat (Java_java_lang_Object_wait__+124)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Object.wait(Native method)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   - waiting on <0x0377a2e9> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$ReferenceQueueDaemon.runInternal(Daemons.java:178)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   - locked <0x0377a2e9> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$Daemon.run(Daemons.java:103)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Thread.run(Thread.java:764)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] "FinalizerDaemon" prio=7 tid=5 Waiting
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d41530 self=0x76d0cc0800
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8960 nice=4 cgrp=default sched=0/0 handle=0x76babe64f0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 2173175 239478 14 ) utm=0 stm=0 core=3 HZ=100
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76baae4000-0x76baae6000 stackSize=1037KB
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait_queue_me+0xe0/0x148
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait+0x104/0x224
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: do_futex+0xe0/0x928
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_futex+0x11c/0x1b0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000001d4ac  /system/lib64/libc.so (syscall+28)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 00000000000e8a94  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+152)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 00000000003cabb8  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+636)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 00000000003cc628  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+376)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 0000000000000aec  /system/framework/arm64/boot.oat (Java_java_lang_Object_wait__JI+140)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Object.wait(Native method)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   - waiting on <0x06c98b6e> (a java.lang.Object)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Object.wait(Object.java:422)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:188)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   - locked <0x06c98b6e> (a java.lang.Object)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:209)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:232)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$Daemon.run(Daemons.java:103)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Thread.run(Thread.java:764)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] "FinalizerWatchdogDaemon" prio=7 tid=6 Sleeping
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d415b8 self=0x76d0cc1200
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8961 nice=4 cgrp=default sched=0/0 handle=0x76baae14f0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 507498 492500 13 ) utm=0 stm=0 core=2 HZ=100
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76ba9df000-0x76ba9e1000 stackSize=1037KB
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait_queue_me+0xe0/0x148
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait+0x104/0x224
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: do_futex+0xe0/0x928
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_futex+0x11c/0x1b0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000001d4b0  /system/lib64/libc.so (syscall+32)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 00000000000e8eb4  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 00000000003cabc8  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+652)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 00000000003cc628  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+376)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 000000000001e3ec  /system/framework/arm64/boot.oat (Java_java_lang_Thread_sleep__Ljava_lang_Object_2JI+188)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Thread.sleep(Native method)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   - sleeping on <0x0160120f> (a java.lang.Object)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Thread.sleep(Thread.java:373)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   - locked <0x0160120f> (a java.lang.Object)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Thread.sleep(Thread.java:314)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$FinalizerWatchdogDaemon.sleepFor(Daemons.java:342)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$FinalizerWatchdogDaemon.waitForFinalization(Daemons.java:364)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$FinalizerWatchdogDaemon.runInternal(Daemons.java:281)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$Daemon.run(Daemons.java:103)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Thread.run(Thread.java:764)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] "HeapTaskDaemon" prio=7 tid=7 WaitingForCheckPointsToRun
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d417e0 self=0x76d0cc4e00
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8962 nice=4 cgrp=default sched=0/0 handle=0x76ba9dc4f0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 122117551 14049217 85 ) utm=10 stm=0 core=4 HZ=100
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76ba8da000-0x76ba8dc000 stackSize=1037KB
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait_queue_me+0xe0/0x148
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait+0x104/0x224
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: do_futex+0xe0/0x928
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_futex+0x11c/0x1b0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000001d4ac  /system/lib64/libc.so (syscall+28)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 00000000000e8a94  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+152)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 00000000000e21d8  /system/lib64/libart.so (art::Barrier::Increment(art::Thread*, int)+84)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 00000000001eac98  /system/lib64/libart.so (art::gc::collector::ConcurrentCopying::RevokeThreadLocalMarkStacks(bool, art::Closure*)+704)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 00000000001eb914  /system/lib64/libart.so (art::gc::collector::ConcurrentCopying::ProcessThreadLocalMarkStacks(bool, art::Closure*)+52)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #05 pc 00000000001eb60c  /system/lib64/libart.so (art::gc::collector::ConcurrentCopying::ProcessMarkStackOnce()+784)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #06 pc 00000000001eb2e4  /system/lib64/libart.so (art::gc::collector::ConcurrentCopying::ProcessMarkStack()+28)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #07 pc 00000000001e52f0  /system/lib64/libart.so (art::gc::collector::ConcurrentCopying::MarkingPhase()+688)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #08 pc 00000000001e3dd4  /system/lib64/libart.so (art::gc::collector::ConcurrentCopying::RunPhases()+1164)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #09 pc 00000000001f8e4c  /system/lib64/libart.so (art::gc::collector::GarbageCollector::Run(art::gc::GcCause, bool)+344)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #10 pc 000000000021c73c  /system/lib64/libart.so (art::gc::Heap::CollectGarbageInternal(art::gc::collector::GcType, art::gc::GcCause, bool)+3352)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #11 pc 000000000022e170  /system/lib64/libart.so (art::gc::Heap::ConcurrentGC(art::Thread*, art::gc::GcCause, bool)+128)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #12 pc 0000000000234784  /system/lib64/libart.so (art::gc::Heap::ConcurrentGCTask::Run(art::Thread*)+40)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #13 pc 0000000000257ee4  /system/lib64/libart.so (art::gc::TaskProcessor::RunAllTasks(art::Thread*)+64)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #14 pc 00000000001f2f2c  /system/framework/arm64/boot-core-libart.oat (Java_dalvik_system_VMRuntime_runHeapTasks__+124)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at dalvik.system.VMRuntime.runHeapTasks(Native method)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$HeapTaskDaemon.runInternal(Daemons.java:461)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Daemons$Daemon.run(Daemons.java:103)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   at java.lang.Thread.run(Thread.java:764)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] "Binder:8952_1" prio=7 tid=8 Native
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d41648 self=0x76c8618400
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8963 nice=0 cgrp=default sched=0/0 handle=0x76ba7d94f0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 4894425 2892761 25 ) utm=0 stm=0 core=2 HZ=100
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76ba6df000-0x76ba6e1000 stackSize=1005KB
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: binder_thread_read+0xf9c/0x10e8
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: binder_ioctl+0x478/0x958
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: do_vfs_ioctl+0x4c4/0x5c0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_ioctl+0x74/0xbc
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000006a4e4  /system/lib64/libc.so (__ioctl+4)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 0000000000023e48  /system/lib64/libc.so (ioctl+136)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 0000000000054938  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+260)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 0000000000054aa8  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 000000000005516c  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #05 pc 000000000007615c  /system/lib64/libbinder.so (???)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #06 pc 00000000000114e8  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+280)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #07 pc 00000000000a9814  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+140)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #08 pc 0000000000067d0c  /system/lib64/libc.so (__pthread_start(void*)+36)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #09 pc 000000000001eba4  /system/lib64/libc.so (__start_thread+68)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   (no managed stack frames)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] "Binder:8952_2" prio=7 tid=9 Native
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d416d0 self=0x76d0cd3e00
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8964 nice=0 cgrp=default sched=0/0 handle=0x76ba6dc4f0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 93732191 5925155 25 ) utm=9 stm=0 core=5 HZ=100
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76ba5e2000-0x76ba5e4000 stackSize=1005KB
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: binder_thread_read+0xf9c/0x10e8
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: binder_ioctl+0x478/0x958
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: do_vfs_ioctl+0x4c4/0x5c0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_ioctl+0x74/0xbc
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000006a4e4  /system/lib64/libc.so (__ioctl+4)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 0000000000023e48  /system/lib64/libc.so (ioctl+136)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 0000000000054938  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+260)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 0000000000054aa8  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 000000000005516c  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #05 pc 000000000007615c  /system/lib64/libbinder.so (???)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #06 pc 00000000000114e8  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+280)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #07 pc 00000000000a9814  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+140)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #08 pc 0000000000067d0c  /system/lib64/libc.so (__pthread_start(void*)+36)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   native: #09 pc 000000000001eba4  /system/lib64/libc.so (__start_thread+68)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   (no managed stack frames)
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528] "Profile Saver" prio=7 tid=10 Native
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x12d41758 self=0x76c862b800
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8965 nice=9 cgrp=default sched=0/0 handle=0x76ba1d74f0
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 795730 11875 8 ) utm=0 stm=0 core=1 HZ=100
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76ba0dd000-0x76ba0df000 stackSize=1005KB
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   | held mutexes=
02-03 17:47:20.682  8952  8952 F zygote64: runtime.cc:528]   kernel: __switch_to+0x94/0xa0
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait_queue_me+0xe0/0x148
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   kernel: futex_wait+0x104/0x224
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   kernel: do_futex+0xe0/0x928
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   kernel: SyS_futex+0x11c/0x1b0
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   kernel: __sys_trace+0x4c/0x50
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   native: #00 pc 000000000001d4b0  /system/lib64/libc.so (syscall+32)
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   native: #01 pc 00000000000e8eb4  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   native: #02 pc 000000000033a524  /system/lib64/libart.so (art::ProfileSaver::Run()+164)
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   native: #03 pc 000000000033d1bc  /system/lib64/libart.so (art::ProfileSaver::RunProfileSaverThread(void*)+92)
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 0000000000067d0c  /system/lib64/libc.so (__pthread_start(void*)+36)
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   native: #05 pc 000000000001eba4  /system/lib64/libc.so (__start_thread+68)
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   (no managed stack frames)
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528] 
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528] "ConnectivityThread" prio=7 tid=12 Native
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   | group="" sCount=1 dsCount=0 flags=1 obj=0x130c0000 self=0x76d0cc6200
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8971 nice=0 cgrp=default sched=0/0 handle=0x76b9f564f0
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   | state=S schedstat=( 2172341 255470 8 ) utm=0 stm=0 core=0 HZ=100
02-03 17:47:20.683  8952  8952 F zygote64: runtime.cc:528]   | stack=0x76b9e54000-0x76b9e56000 stackSize=1037KB
02-03 17:47:20.684  8952  8952 F zygote64: runtime.cc:528]   | sysTid=8984 nice=10 cgrp=default sched=0/0 handle=0x76b72374f0
02-03 17:47:20.684  8952  8952 F zygote64: runtime.cc:528]   native: #04 pc 0000000000000aec  /system/framework/arm64/boot.oat (Java_java_lang_Object_wait__JI+140)
02-03 17:47:20.810  9039  8952 F google-breakpad: Microdump skipped (uninteresting)
02-03 17:47:20.826  8952  8952 W google-breakpad: ### ### ### ### ### ### ### ### ### ### ### ### ###
02-03 17:47:20.826  8952  8952 W google-breakpad: Chrome build fingerprint:
02-03 17:47:20.826  8952  8952 W google-breakpad: 1.6.0
02-03 17:47:20.826  8952  8952 W google-breakpad: 8
02-03 17:47:20.826  8952  8952 W google-breakpad: ### ### ### ### ### ### ### ### ### ### ### ### ###
02-03 17:47:20.829  8952  8952 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 8952 (.main.myandroid), pid 8952 (.main.myandroid)
02-03 17:47:20.940  9043  9043 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstone
02-03 17:47:20.943   610   610 I /system/bin/tombstoned: received crash request for pid 8952
02-03 17:47:20.946  9043  9043 I crash_dump64: performing dump of process 8952 (target tid = 8952)
02-03 17:47:20.946  9043  9043 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
02-03 17:47:20.947  9043  9043 F DEBUG   : Build fingerprint: 'google/bullhead/bullhead:8.1.0/OPP6.171019.012/4453975:user/release-keys'
02-03 17:47:20.947  9043  9043 F DEBUG   : Revision: 'rev_1.0'
02-03 17:47:20.947  9043  9043 F DEBUG   : ABI: 'arm64'
02-03 17:47:20.947  9043  9043 F DEBUG   : pid: 8952, tid: 8952, name: .main.myandroid  >>> com.anu.main.myandroid <<<
02-03 17:47:20.947  9043  9043 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
02-03 17:47:20.954  9043  9043 F DEBUG   : Abort message: 'quick_exception_handler.cc:326] Check failed: callee_method_ != nullptr org.luaj.vm2.LuaValue.invoke'
02-03 17:47:20.954  9043  9043 F DEBUG   :     x0   0000000000000000  x1   00000000000022f8  x2   0000000000000006  x3   0000000000000008
02-03 17:47:20.954  9043  9043 F DEBUG   :     x4   000000775566e000  x5   000000775566e000  x6   000000775566e000  x7   0000000000000000
02-03 17:47:20.954  9043  9043 F DEBUG   :     x8   0000000000000083  x9   0000000010000000  x10  0000007fe14a7310  x11  0000000000000001
02-03 17:47:20.954  9043  9043 F DEBUG   :     x12  0000000000000000  x13  ffffffffa58a1a62  x14  000000001c826fb2  x15  000000003b9aca00
02-03 17:47:20.954  9043  9043 F DEBUG   :     x16  00000055720cffb0  x17  00000077541364b8  x18  0000000000000000  x19  00000000000022f8
02-03 17:47:20.954  9043  9043 F DEBUG   :     x20  00000000000022f8  x21  0000000000000083  x22  00000077557d6a40  x23  0000007fe14a73a9
02-03 17:47:20.954  9043  9043 F DEBUG   :     x24  0000000000000028  x25  0000000000000051  x26  0000000000000044  x27  00000076d0cbeaa0
02-03 17:47:20.954  9043  9043 F DEBUG   :     x28  00000076d0a7f520  x29  0000007fe14a7350  x30  00000077540eb6ec
02-03 17:47:20.954  9043  9043 F DEBUG   :     sp   0000007fe14a7310  pc   00000077540eb714  pstate 0000000060000000
02-03 17:47:20.966  9043  9043 F DEBUG   : 
02-03 17:47:20.966  9043  9043 F DEBUG   : backtrace:
02-03 17:47:20.966  9043  9043 F DEBUG   :     #00 pc 000000000001d714  /system/lib64/libc.so (abort+120)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #01 pc 000000000047ab34  /system/lib64/libart.so (art::Runtime::Abort(char const*)+552)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #02 pc 0000000000572a5c  /system/lib64/libart.so (android::base::LogMessage::~LogMessage()+996)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #03 pc 000000000046f0e0  /system/lib64/libart.so (art::DeoptimizeStackVisitor::FinishStackWalk()+332)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #04 pc 000000000046ec14  /system/lib64/libart.so (art::DeoptimizeStackVisitor::VisitFrame()+84)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #05 pc 0000000000492eb4  /system/lib64/libart.so (_ZN3art12StackVisitor9WalkStackILNS0_16CountTransitionsE0EEEvb+1088)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #06 pc 000000000046e01c  /system/lib64/libart.so (art::QuickExceptionHandler::DeoptimizeStack()+96)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #07 pc 000000000052146c  /system/lib64/libart.so (art::artDeoptimizeImpl(art::Thread*, art::DeoptimizationKind, bool)+608)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #08 pc 0000000000521208  /system/lib64/libart.so (artDeoptimize+16)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #09 pc 000000000055b7c8  /system/lib64/libart.so (art_quick_deoptimize+168)
02-03 17:47:20.966  9043  9043 F DEBUG   :     #10 pc 0000000000032230  /dev/ashmem/dalvik-jit-code-cache (deleted)
02-03 17:47:22.118   610   610 E /system/bin/tombstoned: Tombstone written to: /data/tombstones/tombstone_05
```

**Comment by wanam** [comment]
I'm getting the same issue [here](https://github.com/wanam/YouTubeAdAway/issues/58#issuecomment-362830248).
Robert is already aware of this issue.

**Comment by M66B** [comment]
@wanam should I close this issue or does it contain useful info?

**Comment by wanam** [comment]
I think it's better to keep it open.

**Comment by rovo89** [comment]
Yes, better keep it open. Then once I think I have a solution, you can verify it individually.

**Comment by quorn23** [comment]
Subscribed as i'm suffering from the same issue @rovo89 do you require more logs? (new issue?)

**Comment by rovo89** [comment]
No thanks, I know why this issue occurs. But I need to rethink the whole concept of invalidating possibly outdated method code, which is really not easy.

**Comment by phinney** [comment]
I'm having the same symptom: got a request to deoptimize un-deoptimizable method. Same environment: Android 8.1, but I'm using Nexus 6 .  Sound's like a cause is known, and a solution is tough.  Is there anything known about how to work around the cause? What makes a method un-deoptimizable? Is the cause your finding generic to android or is it specific to the use of eu.faircode.xlua packages.  I'm not using the eu.faircod.xlua packages.  That would at least let me now if my issue is the same related issue.

As guorn23 said if you need any additional logs or other information. I'd be happy to share more details about my situation.

**Comment by phinney** [comment]
Here's what I found out about my situation.  It appears that this error can occur if you don't have the right ClassLoader (whatever right means)  In my situation, I am using an OSGI framework and the framework plays tricks with the ClassLoader; so it easily conceivable that in getting an class from an osgi bundle that android thinks the classloader isn't right and gives this warning.  It probably doesn't do the optimization phase, but other than that, everything seems to be working fine.  So in my case, I think it reasonable to ignore that warning.  I'll look to see if there's anything I can do so that those classes can be optimized.  When I first saw this message in the application log, it was the last entry before my application hung; making me miss associate that error message with the application hanging.  That wasn't the case, I was just making a dumb error of call startActivity when I mean to call startService.

**Comment by rovo89** [comment]
It's hard to explain in a few words. It's not related to a wrong class loader. Deoptimization means that there is compiled code, but it shouldn't be used anymore because the code was compiled under certain assumptions which turned out to be wrong. These assumptions allow for better optimizations, so it's worth to check them at runtime and fall back to the interpreter (and later JIT). In order to deoptimize, certain things about the current execution position, memory registers etc. need to be known.

For those cases that AOSP has, it knows when deoptimization can happen and makes sure to save that information next to the compiled code.

Xposed needs to deoptimize the callers when a method is hooked. This wasn't planned at compile time, so not all required information is available. That's why in the best case, you just see this warning, or in the worst case the app crashes.

I still have to think about better ways. Not deoptimizing is also bad. But alternatives are hard to find.

This should only affect methods which are on the stack when the method is hooked, so hooking methods as early as possible should help. It's also only for compiled code, so resetting the app to interpret-only might help temporarily as well. But there's no real good way to avoid it.

**Comment by M66B** [comment]
I think that was an excellent explanation with lots of detailed info!

**Comment by diggi543** [comment]
And what about fix? I didn`t find any information how to fix it, HELP :(

**Comment by QkiZMR** [comment]
Start using edXposed

**Comment by menshen** [comment]
I I also encountered the same problem in Android8.1.0
@rovo89

**Comment by eebssk1** [comment]
@menshen  Try EdXposed,it also supports android 8.1.The officical xposed for 8.1 has known bugs and it will always be beta.
